matchbox/resource_profile.go

```diff
@@ -156,6 +163,19 @@ func resourceProfileDelete(d *schema.ResourceData, meta interface{}) error {
 	return nil
 }
 
-func containerLinuxConfigName(name string) string {
-	return name + ".yaml.tmpl"
+func containerLinuxConfigName(d *schema.ResourceData) string {
+	filename, _ := containerLinuxConfig(d)
+	return filename
+}
+
+func containerLinuxConfig(d *schema.ResourceData) (filename, config string) {
```

Comment:
Currently `container_linux_config` is Optional, but looks that make sense make it required.

As you can see I updated the PR adding a `validateResourceProfile` that validated that if both options are used or none triggering an error.